src/lib/containers/ForumSearch.tsx

```diff
@@ -0,0 +1,147 @@
+            },
+            props.forumId,
+            accessToken)
+        if (searchResponse.matches.length == 0) {
+            setNoSearchResult(true)
+        }
+        setSearchResult(searchResponse)
+        setMatchList(searchResponse.matches)
+        setIsLoading(false)
+    }
+
+    useEffect(() => {
+        const fetchEntity = async () => {
+            if (props.projectId) {
+                const entity = await getEntity(accessToken, props.projectId)
+                setEntity(entity)
+            }
+        }
+        fetchEntity()
+    }, [])
+
+    const NoSearchResultComponent = () => {
+        return (
+            <div className='text-center'>
+                {NoSearchResults}
+                <Typography variant='body1'>No results with this query</Typography>
+                <Typography variant='body2'>Search the full text of posts, replies, and titles</Typography>
+            </div>
+        )
+    }
+
+    const onLoadMore = async () => {
+        const searchResponse = await forumSearch(
+            {
+                searchString: searchInput,
+                nextPageToken: searchResult?.nextPageToken
+            },
+            props.forumId,
+            accessToken)
+        setSearchResult(searchResponse)
+        if (matchList) {
+            setMatchList([...matchList, ...searchResponse.matches])
+        }
+    }
+
+    return (
+        <div className="bootstrap-4-backport ForumSearch">
+            <div>
+                <FontAwesomeIcon
+                    size={'sm'}
+                    icon={faSearch}
+                    className="SearchIcon"
+                />
+                <input
+                    role='textbox'
+                    type='search'
+                    className={`SearchBar  ${searchResult ? 'SearchBarResult' : ''}`}
+                    placeholder='Search discussions'
+                    value={searchInput}
+                    onChange={event => {
+                        setSearchInput(event.target.value)
+                    }}
+                    onKeyDown={(event: any) => {
+                        if (event.key === 'Enter') {
+                            onSearch()
+                        }
+                    }}
+                />
+                {searchInput && (
+                    <FontAwesomeIcon
+                        size={'sm'}
+                        icon={faTimes}
+                        role="button"
+                        title="Clear Search"
+                        className="ClearSearchIcon"
+                        onClick={() => {
+                            setSearchInput('')
+                        }}
+                    />
+                )}
+            </div>
+            {
+                noSearchResult && <NoSearchResultComponent />
+            }
+            {
+                isLoading ?
+                    <div style={{ textAlign: 'center' }}>
+                        <SynapseSpinner size={40} />
```

Comment:
Instead of a spinner, it would be better to have placeholder Skeleton results.  To do this, I would define and export a "LoadingDiscussionSearchResult" in DiscussionSearchResult, and show a few of those here instead.  See UserCardMedium.LoadingUserCardMedium for example.